src/librustc_mir/transform/inline.rs

```diff
@@ -720,6 +720,9 @@ impl<'a, 'tcx> MutVisitor<'tcx> for Integrator<'a, 'tcx> {
                 }
             }
             TerminatorKind::Unreachable => { }
+            TerminatorKind::FalseEdges { ref mut real_target, .. } => {
+                *real_target = self.update_target(*real_target);
```

Comment:
I think you either need to have a `bug!` here or to also update the imaginary targets.